fluent/fluent_test.go

```diff
@@ -274,116 +284,220 @@ func TestJsonConfig(t *testing.T) {
-		}
-		res.f.Close()
-	case <-time.After(time.Millisecond * 500):
-		t.Error("Async must not block")
+	case <-time.After(duration):
+		t.Fatalf("time out after %s: %s", duration.String(), reason)
+	case <-done:
+		return
 	}
 }
 
-func Test_PostWithTimeNotTimeOut(t *testing.T) {
-	f, err := New(Config{
-		FluentPort:    6666,
-		Async:         false,
-		MarshalAsJSON: true, // easy to check equality
-	})
-	if err != nil {
-		t.Error(err)
-		return
-	}
+func TestCloseOnFailingAsyncConnect(t *testing.T) {
+	t.Skip("Broken tests")
 
-	var testData = []struct {
-		in  map[string]string
-		out string
-	}{
-		{
-			map[string]string{"foo": "bar"},
-			"[\"tag_name\",1482493046,{\"foo\":\"bar\"},{}]",
+	testcases := map[string]Config{
+		"with ForceStopAsyncSend and with RequestAck": {
+			Async:              true,
+			ForceStopAsyncSend: true,
+			RequestAck:         true,
+		},
+		"with ForceStopAsyncSend and without RequestAck": {
+			Async:              true,
+			ForceStopAsyncSend: true,
+			RequestAck:         false,
+		},
+		"without ForceStopAsyncSend and with RequestAck": {
+			Async:              true,
+			ForceStopAsyncSend: false,
+			RequestAck:         true,
 		},
-		{
-			map[string]string{"fuga": "bar", "hoge": "fuga"},
-			"[\"tag_name\",1482493046,{\"fuga\":\"bar\",\"hoge\":\"fuga\"},{}]",
+		"without ForceStopAsyncSend and without RequestAck": {
+			Async:              true,
+			ForceStopAsyncSend: false,
+			RequestAck:         false,
 		},
 	}
-	for _, tt := range testData {
-		conn := &Conn{}
-		f.conn = conn
 
-		err = f.PostWithTime("tag_name", time.Unix(1482493046, 0), tt.in)
-		if err != nil {
-			t.Errorf("in=%s, err=%s", tt.in, err)
-		}
+	for tcname := range testcases {
+		t.Run(tcname, func(t *testing.T) {
+			tc := testcases[tcname]
+			t.Parallel()
 
-		rcv := make([]byte, len(conn.buf))
-		_, _ = conn.Read(rcv)
-		if string(rcv) != tt.out {
-			t.Errorf("got %s, except %s", string(rcv), tt.out)
-		}
+			d := newTestDialer()
+			f, err := newWithDialer(tc, d)
+			if err != nil {
+				t.Errorf("Unexpected error: %v", err)
+			}
 
-		if !conn.writeDeadline.IsZero() {
-			t.Errorf("got %s, except 0", conn.writeDeadline)
-		}
+			timeout(t, 1*time.Second, func() { f.Close() }, "failed to close the logger")
+		})
 	}
 }
 
-func Test_PostMsgpMarshaler(t *testing.T) {
-	f, err := New(Config{
-		FluentPort:    6666,
-		Async:         false,
-		MarshalAsJSON: true, // easy to check equality
-	})
-	if err != nil {
-		t.Error(err)
-		return
+func ackRespMsgp(t *testing.T, ack string) string {
+	msg := AckResp{ack}
+	buf := &bytes.Buffer{}
+	ackW := msgp.NewWriter(buf)
+	if err := msg.EncodeMsg(ackW); err != nil {
+		t.Fatalf("Unexpected error: %v", err)
 	}
+	ackW.Flush()
+	return buf.String()
+}
 
-	var testData = []struct {
-		in  *TestMessage
-		out string
-	}{
-		{
-			&TestMessage{Foo: "bar"},
-			"[\"tag_name\",1482493046,{\"foo\":\"bar\"},{}]",
+func TestCloseOnFailingAsyncReconnect(t *testing.T) {
+	t.Skip("Broken tests")
```

Comment:
To be clear - With #82, we can remove this line, right?

Reply:
I pushed a newer version of #82 with the commit from this PR and a reworked version of my initial fix. As you can see [here](https://github.com/fluent/fluent-logger-golang/pull/82/commits/8ae97ca5c2f1a707583aefe710802fc76eef30aa#diff-008df56d47768ed9d010466fb15f31ac), both `t.Skip()` are removed :slightly_smiling_face:

I'll rebase #82 once this current PR got merged :wink: